### DISCUSSION TITLE
Discussion: Status of the CLR and impact on proposals

### DISCUSSION THREAD

**Discussion by SamPruden** [dotnet/csharplang#255] [General]
1. When can we reasonably expect to see a batch of CLR changes happening?

1. Should a dependency on a CLR change be considered a strong demerit for a feature request?

1. Can those in the know give any guidelines as to generally how the CLR should be factored in when thinking about feature requests?

**Comment by SamPruden** [comment]
Citing #252 as an example of a discussion being hampered by not knowing what's going on with the CLR. I'm coming from the perspective that CLR changes for this feature are unlikely, at least for now. @phi1010 is thinking more ambitiously under the assumption that they can happen.

We're agreed in our goals and differ only on our assumptions, it would be really great if we could have a bit more of an idea about how we should be thinking about this.

**Comment by SamPruden** [comment]
It's a shame to see this go so long without any form of comment, the thumbs indicate that the community considers this a valuable question. Would it be horribly rude or improper of me to say pinging @gafter and/or @MadsTorgersen for something? We're not asking for any kind of official commitment, just some basic guidelines to cite in discussions here.

**Comment by SamPruden** [comment]
And now I see that @gafter just posted #317 a couple of hours before that last comment, which while not directly an answer to this, does make me look like a bit of an idiot. My apologies.

**Comment by gafter** [comment]
LOL!

**Comment by gafter** [comment]
Language changes that only require compiler changes are much, much simpler to do. There are many CLR implementations that would need to support anything that requires CLR changes.
